### DISCUSSION TITLE
AIR needs built in browser and Flash support

### DISCUSSION THREAD

**Discussion by GreyWizard-D13S3L** [airsdk/Adobe-Runtime-Support#1204] [Ideas]
Why not upgrade AIR's WebView and allow it to be used as a truly powerful browser. Google, Safari and Firefox make billions from that.

I do this for Android and iOS. My company's internet site still has full Flash support with this by using my AIR application as a browser.

There are a few lines of code that cause a swf to crash when loaded into AIR instead of Flash, but Harmann could easily upgrade the sdk to ignore those errors.

I use a native plugin in AIR to browse the web, and whenever there is a page that has a swf I hide the browser and use AIR to open the Flash file. Shouldn't be hard to create a desktop browser plugin.

WebView works pretty well, and probably won't need much of an upgrade to open any html page correctly. Not sure why Adobe stopped working on this class.

This is all Harmann needs to add to AIR  to make millions from it. Why abandon all the games and apps that are still all over the internet when AIR has so much more power than Flash did?

This could easily add 20 years of life to AIR, and bring back all the Flash programmers who wish Adobe hadn't killed it off. If all the Flash programmers came back to AIR how many licenses would Harmann sell? That would guarantee AIR becomes stronger.

IMHO AIR is so much more powerful than Html5, and the best 2D cross platform SDK.

**Comment by ajwfrost** [comment]
The challenge, sadly, is in this bit:
> Why not upgrade AIR's WebView

The WebView that's built into desktop AIR implementations is a 10-year-old version of WebKit. If we were to update this to new WebKit or to Chromium etc, then it would lose the support for plug-ins within the browser. It may be possible for some mechanism to try to detect SWF content within a web page, and then render this somehow in AIR, although rendering it within the browser itself (so interacting with the other web page content) would be a challenge I suspect!

Plus it would be a challenge to try to compete with the existing browser offerings - people tend to go with what they're used to or what they're pushed into, hence the prevalence of Chrome (65%) vs Firefox or Edge (3% each) or even Opera (2%) [[ref](https://www.oberlo.com/statistics/browser-market-share)]. And as we saw when people started browsing the web with their iphones: the web adapts to the users, rather than users changing their habits in order to view web pages.

There are other solutions for existing games and apps that are web-based and need Flash Player, for anyone who wanted to still support them, some of these are already in place in the background (and if done right, you shouldn't notice them .... e.g. [CheerpX](https://leaningtech.com/cheerpx-for-flash/)) - much better to have a server-side solution which doesn't require any action by the end user.

Thanks for the suggestion though!

**Comment by Jason0247** [comment]
We have always loved as development. Now we spend more time facing windows and mobile platforms than web pages. I really hope that air can continue. For example, you can also use 64 bit desktop applications in animate...

**Comment by bishpatrik** [comment]
We still have an AIR App that relies on the Acrobat PDF Plugin working in AIRs web view, to be able to view and print PDFs directly from our application.  Unfortunately, presumably due to the age of the Web View this is getting flakier and flakier, and causes several crashes, so in our latest version we have preferences that disable the plugin if it is crashing the users system, and just have a button that opens into the default PDF viewer of the users system, which works but reduces the user experience, so we would like improvements to the webview, to specifically work with the Acrobat Reader plugin.
